File: backend-Nijan/src/main/java/com/ecommerce/backendNijan/service/ContactService.java
package com.ecommerce.backendNijan.service;

import com.ecommerce.backendNijan.model.ContactDto;
import com.ecommerce.backendNijan.response.PageResponse;

import java.util.List;

public interface ContactService {

    ContactDto getContactById(Long contactId);

    void insertContact(ContactDto contact);

    PageResponse<ContactDto> getAllContacts(int pageNo, int pageSize, String sortDirection, String sortBy);
}

File: vite-nijan/src/component/modules/ContactService.js
import http from "./http-common";

class ContactService {

    getAllPageable(page) {
        return http.get('http://localhost:8888/api/contact/list' + "?page_no=" + page.pageNo + "&page_size=" + page.pageSize
        + "&sort_direction=" + page.sortDirection + "&sort_by=" + page.sortBy)
    }

    insert(data) {
        return http.post('http://localhost:8888/api/contact/insert', data);
    }
}
export default new ContactService();